doc(1.7.0): add section for longhornctl (#946)
longhorn/website: content/docs/1.7.0/advanced-resources/longhornctl/_index.md

---
title: Command Line Tool (longhornctl)

> description: Command line tool for Longhorn operations and troubleshooting.

Reviewer: ```suggestion
description: Command line interface (CLI) for Longhorn operations and troubleshooting.
```

weight: 8
---

> Longhorn offers the `longhornctl` command line tool to simplify manual Longhorn operations. This guide covers its functionalities, installation process, and common usage cases.
> 
> ## What is longhornctl?
> 
> `longhornctl` is a remote command line tool for Longhorn. It interacts with Longhorn by creating Kubernetes Custom Resources (CRs) and executing commands inside a dedicated Pod for in-cluster and host operations.

Reviewer: ```suggestion
The `longhornctl` tool is a CLI interface to Longhorn operations. It interacts with Longhorn by creating Kubernetes Custom Resources (CRs) and executing commands inside a dedicated Pod for in-cluster and host operations.
```

> `longhornctl` is a remote command line tool for Longhorn. It interacts with Longhorn by creating Kubernetes Custom Resources (CRs) and executing commands inside a dedicated Pod for in-cluster and host operations.

Reviewer 2: Sentences that describe the contents of a doc file are unnecessary. The first paragraph should immediately provide value to users (in this case, provide a descriptive overview of the tool).

> ## Key Features

Reviewer: ```suggestion
## Common usage scenarios
```

* **Installation:**
  * `longhornctl install preflight`: Perform preflight dependencies installation and setup before installing Longhorn.
* **Operations:**

>   * `longhornctl export replica`: Extract data from a Longhorn replica data directory to a designated directory on its host machine. This can be useful for recovering data when Longhorn is unavailable.

Reviewer: ```suggestion
  * `longhornctl export replica`: Extract data from a Longhorn replica data directory to a designated directory on its host machine. This is useful for recovering data when Longhorn is unavailable.
```

>   * `longhornctl trim volume`: Reclaim unused storage space within a Longhotn volume.

Reviewer: ```suggestion
  * `longhornctl trim volume`: Reclaim unused storage space within a Longhorn volume.
```

> * **Troubleshoots:**

Reviewer: ```suggestion
* **Troubleshooting:**
```

> * **Installation:**
>   * `longhornctl install preflight`: Perform preflight dependencies installation and setup before installing Longhorn.
> * **Operations:**
>   * `longhornctl export replica`: Extract data from a Longhorn replica data directory to a designated directory on its host machine. This can be useful for recovering data when Longhorn is unavailable.
>   * `longhornctl trim volume`: Reclaim unused storage space within a Longhotn volume.
> * **Troubleshoots:**
>   * `longhornctl check preflight`: Identifies potential issues before encountering problems during Longhorn usage.
>   * `longhornctl get replica`: Retrieve details about Longhorn replicas on the host.

Reviewer 2: ```suggestion
- **Installation**
  - `longhornctl install preflight`: Installs and sets up preflight dependencies before Longhorn is installed.
- **General Operations**
  - `longhornctl export replica`: Extracts data from a Longhorn replica data directory to a designated directory on its host machine. You can use this feature to recover data when Longhorn is unavailable.
  - `longhornctl trim volume`: Reclaims unused storage space within a Longhotn volume.
- **Troubleshooting**
  - `longhornctl check preflight`: Identifies issues that may occur during Longhorn usage.
  - `longhornctl get replica`: Retrieves details about Longhorn replicas on the host.
```

## Usage

> For detailed information on available commands and their usage, refer to the [GitHub document](https://github.com/longhorn/cli/tree/{{< current-version >}}/docs/longhornctl.md), or run `longhornctl help` within your terminal.

Reviewer 2: ```suggestion
For more information about the available commands, see [this document](https://github.com/longhorn/cli/tree/{{< current-version >}}/docs/longhornctl.md) in the GitHub repository, or run `longhornctl help` in your terminal.
```
